stl/inc/algorithm

```diff
@@ -221,6 +247,17 @@ template <class _Iter, class _Ty1, class _Ty2>
 constexpr bool _Vector_alg_in_ranges_replace_is_safe =
     _Vector_alg_in_replace_is_safe<_Iter, _Ty1> // can search and replace
     && _Vector_alg_in_find_is_safe_elem<_Ty2, _Iter_value_t<_Iter>>; // replacement fits
+
+// Can we activate the vector algorithms for unique?
+template <class _Iter, class _Pr>
+constexpr bool _Vector_alg_in_unique_is_safe =
+    _Iterator_is_contiguous<_Iter> && !_Iterator_is_volatile<_Iter> // Contiguous nonvolatile iterator
+    && _Is_any_of_v<_Pr,
+#if _HAS_CXX20
+        _RANGES equal_to,
+#endif // _HAS_CXX20
+        _STD equal_to<>> // default comparison

```

Comment:
This rejects `equal_to<T>`.

Would `_Can_memcmp_elements_with_pred` be applicable here? It feels like we've already solved this problem.

Reply:
That seems using too generic tool for simpler problem (we have only one iterator type here), but still looks like a good idea.